apps/studio/src/server/modules/resource/__tests__/resource.service.test.ts

```diff
@@ -856,6 +890,403 @@ describe("resource.service", () => {
+      })
+      await db
+        .updateTable("Blob")
+        .where("id", "=", pageBlob.id)
+        .set({
+          content: {
+            ...pageBlob.content,
+            page: {
+              image: {
+                src: "https://pageblob.com",
+                alt: "im a page blob image alt text",
+              },
+            },
+          },
+        })
+        .execute()
+
+      const { folder } = await setupFolder({
+        title: "Folder A",
+        permalink: "folder-a",
+        siteId: site.id,
+        parentId: parentFolder.id,
+        state: ResourceState.Published,
+      })
+
+      const { blob: folderAIndexPageBlob } = await setupPageResource({
+        title: "Folder A",
+        resourceType: ResourceType.IndexPage,
+        siteId: site.id,
+        parentId: folder.id,
+        state: ResourceState.Published,
+        userId: (await setupUser({})).id,
+      })
+      await db
+        .updateTable("Blob")
+        .where("id", "=", folderAIndexPageBlob.id)
+        .set({
+          content: {
+            ...folderAIndexPageBlob.content,
+            page: {
+              contentPageHeader: {
+                summary: "Hello im the index page",
+              },
+              image: {
+                src: "https://indexpageblob.com",
+                alt: "im a index page blob image alt text",
+              },
+            },
+          },
+        })
+        .execute()
+
+      const { collection } = await setupCollection({
+        title: "Collection A",
+        permalink: "collection-a",
+        siteId: site.id,
+        parentId: parentFolder.id,
+        state: ResourceState.Published,
+      })
+
+      // Act
+      const result = await getLocalisedSitemap(site.id, Number(indexPage.id))
+
+      // Assert
+      const children = result.children?.at(0)?.children
+      expect(children?.length).toBe(3)
+
+      // Assert: Find Page in the sitemap
+      const pageNode = result.children
+        ?.at(0)
+        ?.children?.find((child) => child.id === page.id)
+      expect(pageNode?.title).toBe(page.title)
+      expect(pageNode?.image?.src).toBe("https://pageblob.com")
+
+      // Assert: Find Folder in the sitemap
+      const folderNode = result.children
+        ?.at(0)
+        ?.children?.find((child) => child.id === folder.id)
+      expect(folderNode?.title).toBe(folder.title)
+      expect(folderNode?.summary).toBe("Hello im the index page")
+      expect(folderNode?.image?.src).toBe("https://indexpageblob.com")
+
+      // Assert: Find Collection in the sitemap
+      const collectionNode = result.children
+        ?.at(0)
+        ?.children?.find((child) => child.id === collection.id)
+      expect(collectionNode?.title).toBe(collection.title)
+      expect(collectionNode?.summary).toBe(`Pages in ${collection.title}`)
+      expect(collectionNode?.image?.src).toBeUndefined()
+    })
+
+    it("should include any nested collections if resourceId is a RootPage", async () => {
```

Comment:
all tests other than this was from https://github.com/opengovsg/isomer/pull/1363